### PR TITLE
Switch from Golang to Python

### LINKED ISSUE #38: Investigate using Python for statistical algorithms
Transitioning from Go to Python could open up many more algorithms for the PHPA to use without having to reimplement in Go.

Suggested in this comment: https://github.com/jthomperoo/predictive-horizontal-pod-autoscaler/issues/27#issuecomment-696585342

### PR DESCRIPTION
**Pull request by jthomperoo**
Resolves #38

### CODE REVIEW

**Comment by kevjin** [comment]
hey @jthomperoo! I just saw this, and I'm curious what other TODOs do you need to get this merged in? or is it just like final/edge case testing? thanks!

**Comment by jthomperoo** [comment]
Hey @kevjin - thanks for your other pull request btw, just having a look through now and I'll get it merged in - for this work to be honest I had left it as I didn't have much time. As far as I can remember it was 90% done, just wanted to test that it worked without breaking anything. Thanks for reminding me of this, I think I'll have a look at these changes at the weekend and see if I can get it merged in. Feel free to have a look through the changes. 

There was a request to add ARIMA forecasting to this (#37) which was on the to-do list after this Python stuff was added in, which would hopefully be easy enough with the Python libraries (easier than reimplementing in Golang anyway!).

**Comment by kevjin** [comment]
awesome thanks! I'm actually trying to get this working for my own LSTM model haha, so I'll definitely be looking, and doing some of my own manual testing on this PR. I'll let you know if I find any bugs!

File: algorithm/algorithm.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package algorithm
+
+import (
+	cpaconfig "github.com/jthomperoo/custom-pod-autoscaler/config"
+	"github.com/jthomperoo/custom-pod-autoscaler/execute"
+)
+
+const (
+	entrypoint   = "python"
+	shellTimeout = 10000
```
Comment (kevjin) [inline algorithm/algorithm.go:10]: imo a hard 10 second timeout is a little too restrictive. Some models, like ARIMA, will probably take longer to run. Also the more `storedValues` you have, the longer it'll take the model to run. What about 30 seconds?

Although, I'm not sure if the CPA would respect the longer timeout (haven't really looked at it)

Comment (jthomperoo) [reply algorithm/algorithm.go]: Yep good point, I've made a bunch of changes so that this value is configurable, and upped the default timeout to 30s.

**Pull request merged**